### PR TITLE
Prevented theme swap at startup in daily mode

### PR DESCRIPTION
**Pull request by Denis-net**
Based on what we said on the issue #19: 
I added a note about it and fixed the  `randomThemeSwitcher.switchMode` "daily",
because even when set themes were swapped at startup.

Happy new year ! 🥂🍾🎉🎊

### CODE REVIEW

**Comment by Denis-net** [comment]
Good evening !

I have also added a new feature: Detect when the theme changes and if ! present ask for adding it.
I refactored a bit the already present code and i think it's a good result.

The only drawback is that it's not workspace friendly: if you keep a local `randomThemeSwitcher.themeList` it will ask user to add a theme even if it's already in the `%appdata%/Code/User/settings.json` because 
it reads from the `/workspaceFolder/.vscode/settings.json` but write into the `%appdata%/Code/User/settings.json` (i.e. it will left the former untouched).
But this is not a big issue.

File: src/extension.ts [hunk 1]
```diff
@@ -116,11 +122,18 @@ async function getThemeList(
...
-
   const currentThemeName: string | undefined = userSettings.get('workbench.colorTheme');
 
+  _addCurrentTheme(themeList, currentThemeName);
+}
+
+function _addCurrentTheme(themeList: string[], currentThemeName: string | string[] | undefined) {
   if (typeof currentThemeName !== 'undefined') {
-    saveThemes(themeList.concat(currentThemeName).sort(), Messages.AddedTheme(currentThemeName));
+    let keys = typeof currentThemeName === typeof Array ?
```
Comment (zanza00) [inline src/extension.ts]: Since here you are not modifying the variable in can be a `const`

File: src/enums.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ export class Messages {
 
 
 export class ThemeTypes {
-    public static Both = 'both';
-    public static Ligth = 'vs';
-    public static Dark = 'vs-dark';
+    public static Both: { key: string, value: string } = { key: 'Both', value: 'both', };
+    public static Ligth: { key: string, value: string } = { key: 'Ligth', value: 'vs', };
+    public static Dark: { key: string, value: string } = { key: 'Dark', value: 'vs-dark', };
```
Comment (zanza00) [inline src/enums.ts]: Why this change? You are using it in a way that I don't immediately understand.

File: src/extension.ts [hunk 3]
```diff
@@ -163,10 +176,26 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
       if (e.affectsConfiguration('workbench.colorTheme')) {
         const userSettings = getUserSettings();
         const currentTheme = userSettings.get('workbench.colorTheme', '');
-        const oneMoreTimeThemeList: string[] = <string[]>userSettings.get(SettingsKeys.PreventReloadThemeList, MATERIAL_LIST);
 
+        const oneMoreTimeThemeList: string[] = <string[]>userSettings.get(SettingsKeys.PreventReloadThemeList, MATERIAL_LIST);
         context.globalState.update(LAST_THEME_NEEDS_TO_PERSIST, oneMoreTimeThemeList.findIndex(mat => mat === currentTheme) !== -1);
         themeList = await getThemeList(getExtensionConfig(), getUserSettings());
+
+        if (!themeList.includes(currentTheme)) {
```
Comment (zanza00) [inline src/extension.ts]: Are you sure that here it doesn't trigger when the current theme is changed by the extension? Like if someone changes by using the command prompt or the hotkey?

File: issues/19/readme.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# The issue
```
Comment (zanza00) [inline issues/19/readme.md]: What is the meaning of this file, I never see something like this.

File: src/extension.ts (hunk 1 above)
Comment (Denis-net) [reply src/extension.ts]: Good evening !
Thanks,
I work in a office so we don't use GitHub in this way;
How can i request you as a reviewer ?
Is there a button or something ?

Regarding the issue: Fixed !👍🏼
(i missed it while refactoring)

File: src/enums.ts (hunk 2 above)
Comment (Denis-net) [reply src/enums.ts]: Oh yes: when i added it i was modelling it as a key-value pair, but the right terms would be `display_member` and `value_member`.
This will improve the user experience, 
because if an user reads:
"Both"
"Ligth"
"Dark"
can definitley better understand the question, instead of reading: (user reactions in parens)
"both" (?)
"vs" (?????)
"vs-dark" (ohhh, now i think i can understand what the question may refers to).

I'm not sure if a better code structure is available;
Anyway, I remove the last trailing commas 👍🏼

File: src/extension.ts (hunk 3 above)
Comment (Denis-net) [reply src/extension.ts]: I have thought it could be useful for the user when setting a theme to know if it's not in the list and providing a button in order to add it.
Moreover i don't know any vscode event that informs the extensions about new themes/extensions that are installed.
(maybe looking at the extension folder or setting an interval could do the trick, but i preferred this approach)

I could turn the `await` into a `thenable` in order to speed the things up.

Comment (Denis-net) [reply src/extension.ts]: I fixed it, then i discovered another bug :
getThemeList() returns a themeList without the current theme, so the above condition
``` Typescript
if (!themeList.includes(currentTheme)) {
```
get always true and the popup always appears.

A good side effect of having fixed it is that now the swap is more accurate (because previously it will skip only the first "current theme" while considering the next "current theme", if you take a closer look to the next commit you can see the hidden issue that got solved) 
Fixed it i discovered another little bug:
themeList used by the `changeTheme` could resulting being a little behind the actual themeList settings (i.e. if the user adds or removes themes from the list he has to reload VSC in order to get the new themeList configuration ready, check the _themeList in the next commit to spot the issue)

File: issues/19/readme.md (hunk 4 above)
Comment (Denis-net) [reply issues/19/readme.md]: Nevermind, it's just me:
sometimes i keep important issues in a folder called issue, because it contains useful information that can be consulted when needed.
I'll remove it.👍🏼

**Comment by Denis-net** [comment]
Thank you for the invitation ! ❤
It's a real honor to be part of this !

Ok, i made a commit reflecting the changes.
Now, i can go to sleep 😅
Good evening ! 😊✌🏼

File: src/enums.ts (hunk 2 above)
Comment (zanza00) [reply src/enums.ts]: Now I get it, it,s the similar concept of a dropdown menù, the user see a pretty string, for the program internals the id is better.

File: src/extension.ts (hunk 1 above)
Comment (zanza00) [reply src/extension.ts]: here is the official documentation that can explain it better than me https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/requesting-a-pull-request-review

**Comment by zanza00** [comment]
@Denis-net welcome on board :)

File: src/extension.ts (hunk 3 above)
Comment (zanza00) [reply src/extension.ts]: this could be bad for consistency. For now it's good as it is.

Comment (Denis-net) [reply src/extension.ts]: Not sure what are you referring to:

1) I fixed the issue with the themeList That doesn't contain the currentTheme 
2) the fix automatically improve the random theme swap accuracy
3) i noticed that the themeList used in theme swap could potentially not being aligned with the other functions list

I guess the 3, because the first two could be more difficult to revert 😅

Anyway, i have learned how to request a review, so in the next commit i'll know what to do.
Thanks 😊✌🏼
